src/Platform/Microsoft.Testing.Platform/CommandLine/InformativeCommandLineTestHost.cs

```diff
@@ -2,12 +2,31 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using Microsoft.Testing.Platform.Hosts;
+using Microsoft.Testing.Platform.IPC;
 
 namespace Microsoft.Testing.Platform.CommandLine;
 
-internal sealed class InformativeCommandLineTestHost(int returnValue) : ITestHost
+internal sealed class InformativeCommandLineTestHost(int returnValue, NamedPipeClient? namedPipeClient = null) : ITestHost, IDisposable
+#if NETCOREAPP
+#pragma warning disable SA1001 // Commas should be spaced correctly
+    , IAsyncDisposable
+#pragma warning restore SA1001 // Commas should be spaced correctly
+#endif
 {
     private readonly int _returnValue = returnValue;
+    private readonly NamedPipeClient? _namedPipeClient = namedPipeClient;
 
     public Task<int> RunAsync() => Task.FromResult(_returnValue);
+
+    public void Dispose() => _namedPipeClient?.Dispose();
```

Comment:
include this in the else like above if would be good do not compile at all the non async version for core so we do only async disposal where we have support

Reply:
never mind me keep as-is maybe we will have some context where we cannot go async...the disposal helper will take the correct decision https://github.com/microsoft/testfx/blob/main/src/Platform/Microsoft.Testing.Platform/Helpers/DisposeHelper.cs